AppCenter/AppCenterTests/MSHttpClientTests.m

```diff
@@ -0,0 +1,552 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "AppCenter+Internal.h"
+#import "MSAppCenterErrors.h"
+#import "MSDevice.h"
+#import "MSDeviceInternal.h"
+#import "MSHttpClient.h"
+#import "MSHttpIngestionPrivate.h"
+#import "MSHttpTestUtil.h"
+#import "MSIngestionCall.h"
+#import "MSIngestionDelegate.h"
+#import "MSMockLog.h"
+#import "MSTestFrameworks.h"
+#import <OHHTTPStubs/NSURLRequest+HTTPBodyTesting.h>
+#import <OHHTTPStubs/OHHTTPStubs.h>
+
+static NSTimeInterval const kMSTestTimeout = 5.0;
+
+@interface MSHttpClientTests : XCTestCase
+
+@property(nonatomic) MSHttpClient *sut;
+@property(nonatomic) id reachabilityMock;
+@property(nonatomic) NetworkStatus currentNetworkStatus;
+
+@end
+
+@interface MSHttpClient ()
+
+- (instancetype)initWithMaxHttpConnectionsPerHost:(NSNumber *)maxHttpConnectionsPerHost
+                                   retryIntervals:(NSArray *)retryIntervals
+                                     reachability:(MS_Reachability *)reachability;
+
+@end
+
+@implementation MSHttpClientTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Mock reachability.
+  self.reachabilityMock = OCMClassMock([MS_Reachability class]);
+  self.currentNetworkStatus = ReachableViaWiFi;
+  OCMStub([self.reachabilityMock currentReachabilityStatus]).andDo(^(NSInvocation *invocation) {
+    NetworkStatus test = self.currentNetworkStatus;
+    [invocation setReturnValue:&test];
+  });
+}
+
+- (void)tearDown {
+  [super tearDown];
+
+  [MSHttpTestUtil removeAllStubs];
+  [self.reachabilityMock stopMocking];
+
+  /*
+   * Setting the variable to nil. We are experiencing test failure on Xcode 9 beta because the instance that was used for previous test
```

Comment:
Is it a good time to consider removing `self.sut` in this file and use local variable in each test? This will allow us to run all unit tests in parallel to speed up.